server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3994,7 +4004,12 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
                 }
                 // update password for this host
                 final DetailVO nv = _detailsDao.findDetail(host.getId(), ApiConstants.USERNAME);
-                if (nv.getValue().equals(cmd.getUsername())) {
+                if (nv == null) {
+                    final DetailVO nvu = new DetailVO(host.getId(), ApiConstants.USERNAME, cmd.getUsername());
```

Comment:
@ravening  Username null or empty fix is missing here.

Reply:
@sureshanaparti fixed it